test/unittests/test_xmlupload/test_upload_stash_with_mock.py

```diff
@@ -0,0 +1,184 @@
+
+    def get_token(self) -> str:
+        raise AssertionError("'get token' not implemented in this mock")
+
+    def login(self, email: str, password: str) -> None:
+        raise AssertionError("'login' not implemented in this mock")
+
+    def logout(self) -> None:
+        raise AssertionError("'logout' not implemented in this mock")
+
+
+class TestUploadLinkValueStashes:
+    def test_upload_link_value_stash(self) -> None:
+        """Upload stashed link values (resptr), if all goes well."""
+        stash = Stash.make(
+            standoff_stash=None,
+            link_value_stash=LinkValueStash.make(
+                [
+                    LinkValueStashItem("001", "sometype", "someprop", "002"),
+                ],
+            ),
+        )
+        assert stash
+        id2iri_mapping = {
+            "001": "http://www.rdfh.ch/0001/001",
+            "002": "http://www.rdfh.ch/0001/002",
+        }
+        con: Connection = ConnectionMock(
+            get_responses=[{"@context": {}}],
+            post_responses=[{}],
+        )
+        nonapplied = _upload_stash(
+            stash=stash,
+            id2iri_mapping=id2iri_mapping,
+            con=con,
+            verbose=False,
+        )
+        assert nonapplied is None
+
+
+class TestUploadTextValueStashes:
+    def test_upload_link_value_stash(self) -> None:
+        """Upload stashed text values (standoff), if all goes well."""
+        value_uuid = str(uuid4())
+        property_name = "someprop"
+        stash = Stash.make(
+            standoff_stash=StandoffStash.make(
+                [
+                    StandoffStashItem(
+                        "001", "sometype", value_uuid, property_name, KnoraStandoffXml("<p>some text</p>")
+                    ),
+                ]
+            ),
+            link_value_stash=None,
+        )
+        assert stash
+        id2iri_mapping = {
+            "001": "http://www.rdfh.ch/0001/001",
+            "002": "http://www.rdfh.ch/0001/002",
+        }
+        con: Connection = ConnectionMock(
+            get_responses=[
+                {
+                    property_name: [
+                        {
+                            "@id": "http://www.rdfh.ch/0001/001/values/01",
+                            "knora-api:textValueAsXml": "<p>not relevant</p>",
+                        },
+                        {
+                            "@id": "http://www.rdfh.ch/0001/001/values/01",
+                            "knora-api:textValueAsXml": f"<p>{value_uuid}</p>",
+                        },
+                    ],
+                    "@context": {},
+                },
+            ],
+            put_responses=[{}],
+        )
+        nonapplied = _upload_stash(
+            stash=stash,
+            id2iri_mapping=id2iri_mapping,
+            con=con,
+            verbose=False,
+        )
+        assert nonapplied is None
+
+    def test_not_upload_link_value_stash_if_uuid_not_on_value(self) -> None:
```

Comment:
Not sure about your terminology, but perhaps you mean:

```suggestion
    def test_not_upload_text_value_stash_if_uuid_not_on_value(self) -> None:
```